flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/codegen/ExpressionReducer.scala

```diff
@@ -100,7 +100,11 @@ class ExpressionReducer(
       case _ => throw new TableException("RichMapFunction[GenericRow, GenericRow] required here")
     }
 
-    val parameters = if (config.getConf != null) config.getConf else new Configuration()
+    val parameters = if (config.getConfiguration != null) {
```

Comment:
According to the `TableConfig` a config can never be null. Introducing arbitrary null checks is not helpful.